src/autolabel/dataset_loader.py

```diff
@@ -199,3 +202,32 @@ def _read_file(
             )
         else:
             raise ValueError(f"Unsupported file format: {file}")
+
+    def _read_hf_dataset(
+        self,
+        dataset: Dataset,
+        config: AutolabelConfig,
+        max_items: int = None,
+        start_index: int = 0,
+    ) -> None:
+        """Read the huggingface dataset and sets dat, inputs and gt_labels
+
+        Args:
+            dataset (Dataset): dataset object to read from
+            config (AutolabelConfig): config object
+            max_items (int, optional): max number of items to read. Defaults to None.
+            start_index (int, optional): start index to read from. Defaults to 0.
+        """
+        self.dat = pd.DataFrame(
+            dataset[
+                start_index : max_items if max_items and max_items > 0 else len(dataset)
```

Comment:
do we need to do any sanity checks on the dataset object before attempting to convert to dataframe?

Reply:
[This](https://discuss.huggingface.co/t/converting-an-hf-dataset-to-pandas/12043) seems to be a more robust solution so I'll implement it here as well